superset/models/core.py

```diff
@@ -347,7 +347,7 @@ def url(self):
             json_metadata = json.loads(self.json_metadata)
             default_filters = json_metadata.get('default_filters')
             # make sure default_filters is not empty
-            if default_filters and json.loads(default_filters):
+            if default_filters and default_filters != '{}' :
```

Comment:
do you think that this will suffice? it shouldn't be a try catch?

Reply:
I saw production data that contains bad default_filters (invalid json string), and it breaks at `json.loads`. 

I am thinking if user pass in invalid string, dashboard will not load default filters correctly. The bad data will be handled in dashboard js code. Here is just to generate a url (with preloaded_filter parameter), and this url parameter will not affect dashboard initial state. 

So I would allow invalid json string in it. But i don't want to generate an empty url parameter like:
http://localhost:8080/superset/dashboard/7/?preselect_filters={}

Reply:
hmm, but what's the advantage of passing invalid data to the frontend vs a `try/catch` here, and simply not passing anything when it's invalid?

Reply:
added try/catch.